time_smoother/time.go

```diff
@@ -0,0 +1,111 @@
+package time_smoother
+
+import (
+	"math"
+	"time"
+
+	"github.com/dterei/gotsc"
+)
+
+type TimeSmoother struct {
+	lastRdtsc      uint64 // last cycle count
+	lastRdtscNanos uint64 // last cycle-synced time
+	lastSystemTime uint64 // last system time
+	rdtscTime      uint64 // time to call rdtsc
+
+	clockSpeed          float64
+	clockSpeedDecayRate float64
+
+	adjustedTimeRate float64 // nanos per clock
+
+	syncInterval uint64
+
+	stop chan bool
```

Comment:
If `stop` is only acting as a signal, an empty struct would be more memory efficient to use instead of a bool. 

Here's a [blog post by Dave Cheney](https://dave.cheney.net/2014/03/25/the-empty-struct) explaining why.